src/main/java/org/opensearch/neuralsearch/processor/util/ProcessorUtils.java

```diff
@@ -0,0 +1,182 @@
+     * @return The status indicating that the SearchHits are in correct form to perform the Rerank
+     */
+    public static boolean validateRerankCriteria(
+        SearchHit[] searchHits,
+        SearchHitValidator validator,
+        ActionListener<List<Float>> listener
+    ) {
+        for (SearchHit hit : searchHits) {
+            try {
+                validator.validate(hit);
+            } catch (IllegalArgumentException e) {
+                listener.onFailure(e);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Used to get the numeric mapping from the sourcemap using the <code>target_field</code>
+     * <hr>
+     * <b>This method assumes that the path to the mapping exists (and is numerical) as checked by {@link #validateRerankCriteria(SearchHit[], SearchHitValidator, ActionListener)}</b>
+     * As such no error checking is done in the methods implementing this functionality
+     * @param sourceAsMap the map of maps that contains the <code>targetField</code>
+     * @param targetField the path to take to get the score to replace by
+     * @return The numerical score found using the <code>target_field</code>
+     */
+    public static float getScoreFromSourceMap(Map<String, Object> sourceAsMap, String targetField) {
+        Object val = getValueFromSource(sourceAsMap, targetField).get();
+        return ((Number) val).floatValue();
+    }
+
+    /**
+     * This method performs the deletion of the targetField and emptyMaps in 3 phases
+     * <ol>
+     *     <li>Collect the maps and the respective keys (the key is used to get the inner map) in a stack. It will be used
+     *     to delete empty maps and the target field</li>
+     *     <li>Delete the top most entry, this is guaranteed even when the source mapping is non nested. This is the
+     *     mapping containing the targetField</li>
+     *     <li>Iteratively delete the rest of the maps that have (possibly been) emptied as the result of deleting the targetField</li>
+     * </ol>
+     * <hr>
+     * <b>This method assumes that the path to the mapping exists as checked by {@link #validateRerankCriteria(SearchHit[], SearchHitValidator, ActionListener)}</b>
+     * As such no error checking is done in the methods implementing this functionality
+     * @param sourceAsMap the map of maps that contains the <code>targetField</code>
+     * @param targetField The path to take to remove the targetField
+     * @implNote You can think of this algorithm as a recursive one the base case is deleting the targetField. The recursive case
+     * is going to the next map along with the respective key. Along the way if it finds a map is empty it will delete it
+     */
+    public static void removeTargetFieldFromSource(Map<String, Object> sourceAsMap, String targetField) {
+        Stack<Tuple<Map<String, Object>, String>> parentMapChildrenKeyTupleStack = new Stack<>();
+        String[] keys = targetField.split("\\.");
```

Comment:
what if a key contains the special character "."?

Reply:
I didnt know that was possible the expected behavior would still split it and return a exception since it could not find it

Reply:
it's possible to have dot in field name, try this use case

```
PUT /sample_index/_doc/1
{
  "first.name": "Example",
  "price": 29.99,
  "description": "To be or not to be, that is the question"
}
GET  /sample_index/_doc/1
{
  "_index": "sample_index",
  "_id": "1",
  "_version": 1,
  "_seq_no": 0,
  "_primary_term": 1,
  "found": true,
  "_source": {
    "first.name": "Example",
    "price": 29.99,
    "description": "To be or not to be, that is the question"
  }
}
```

Reply:
if you are not using JsonPath, this is an example in the IngestDocument which has a `HasField` method https://github.com/opensearch-project/OpenSearch/blob/456ca97f077929ba51d53dc8578b9b770e3a2417/server/src/main/java/org/opensearch/ingest/IngestDocument.java#L233

Reply:
I think for now i will leave it how it is currently but I do see that as an enhancement simply because in the [doc pr](https://github.com/opensearch-project/documentation-website/pull/8593) I stated how the user should use the argument